src/utils/compile.ts

```diff
@@ -106,3 +131,119 @@ export const compile = async (
         throw new CompilationFailedError(error as string);
     }
 };
+
+const getWithoutFileExtension = (path: string) => path.slice(0, path.lastIndexOf("."));
+
+const getVirtualEntry = (entryFileNames: string[]): string => {
+    const filePathMap = entryFileNames.map((filePath) => {
+        const fileName = filePath.split("/").pop();
+        if (!fileName) {
+            throw new CompilationFailedError("No filename provided");
+        }
+
+        return {
+            name: getWithoutFileExtension(fileName),
+            path: getWithoutFileExtension(filePath.replace(/^(\.\/)/, "")), // Get relative path without leading ./
+        };
+    });
+
+    return `data:text/typescript,${filePathMap
+        .map(({ name, path }) => `import ${name} from './${path}';`)
+        .join("")}export { index, settings };`;
+};
+
+const webpackCompile = async (
+    projectPath: string,
+    entryFileNames: string[],
+    iifeGlobalName: string,
+    options: CompilerOptions,
+): Promise<void> => {
+    const compiler = webpack({
+        mode: options.env?.NODE_ENV === "development" ? "development" : "production",
+        context: projectPath,
+        externals: {
+            react: "React",
+            "react-dom": "ReactDOM",
+        },
+        entry: {
+            "./index.tsx": getVirtualEntry(entryFileNames),
```

Comment:
```suggestion
            "./_virtual/index.tsx": getVirtualEntry(entryFileNames),
```
To be extra explicit about it